lib/Varien/File/Uploader.php

```diff
@@ -505,8 +505,8 @@ private function _setUploadFileId($fileId)
             preg_match("/^(.*?)\[(.*?)\]$/", $fileId, $file);
 
             if (count($file) > 0
-                && (count($file[0]) > 0)
-                && (count($file[1]) > 0)
+                && isset($file[0])
+                && isset($file[1])
```

Comment:
@colinmollenhour suggestion is even better:

```suggestion
            if (preg_match('/^(\w+)\[(\w+)\]$/', $fileId, $file)) {
```